### PR TITLE
[editor] Add variable substitution in editor v2

### PR DESCRIPTION
**Pull request by JohanAhlen**
First iteration, still need to add the different input types.

For date picker we'll need to wait for Vue 3.

### CODE REVIEW

File: desktop/core/src/desktop/js/apps/editor/components/variableSubstitution/VariableSubstitution.vue [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  ): Promise<void> => {
+    try {
+      variable.path = entry.getQualifiedPath();
+      variable.catalogEntry = entry;
+
+      const sourceMeta = <FieldSourceMeta>await entry.getSourceMeta({ silenceErrors: true });
+
+      const colType = (sourceMeta && sourceMeta.type) || 'string';
+
+      switch (colType) {
```
Comment (romainr) [inline desktop/core/src/desktop/js/apps/editor/components/variableSubstitution/VariableSubstitution.vue:81]: Nice!

File: desktop/libs/notebook/src/notebook/templates/editor2.mako [hunk 2]
```diff
@@ -382,56 +382,58 @@
...
-                event: { 'keydown': $parent.onKeydownInVariable }
-              "></select>
-          <!-- /ko -->
-        </div>
-      </li>
-    </ul>
-  </div>
-</script>
+
+## TODO: Move additional types to VariableSubstitution.vue
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/templates/editor2.mako:386]: Drop?

**Review by romainr — APPROVED** [review]
Test wise we have the formats in the Editor help, e.g. ${limit}, ${limit=10}
and :+1: for datepicker (we can add the date format in the dialect properties at some point)

File: desktop/libs/notebook/src/notebook/templates/editor2.mako (hunk 2 above)
Comment (JohanAhlen) [reply desktop/libs/notebook/src/notebook/templates/editor2.mako:386]: Keeping it as a reference for now, will re-visit after predict...

**Pull request merged**